src/app.tsx

```diff
@@ -129,230 +253,217 @@ export const App = () => {
-                            storagePools={(storagePools || []).filter(pool => pool && pool.connectionName == connectionName)}
-                            networks={(networks || []).filter(network => network && network.connectionName == connectionName)}
-                            nodeDevices={(nodeDevices || []).filter(device => device && device.connectionName == connectionName)}
-                            key={vmId(vm.name)}
-                        />
-                    </>
-                );
-            return expandedContent;
-        }
+    return (
+        <>
+            {getInlineNotifications()}
+            <HostVmsList
+                vms={vms}
+                ui={ui}
+                storagePools={storagePools}
+                networks={networks}
+                actions={vmActions}
+            />
+        </>
+    );
+};
 
-        const loggedUser = systemInfo.loggedUser;
+const AppVM = ({
+    consoleCardStates,
+    name,
+    connection,
+} : {
+    consoleCardStates: ConsoleCardStates,
+    name: string,
+    connection: ConnectionName,
+}) => {
+    const { vms, config, storagePools, systemInfo, ui, networks, nodeDevices } = store.getState();
+    const combinedVms = [...vms, ...dummyVmsFilter(vms, ui.vms)];
 
+    const vm = combinedVms.find(vm => vm.name == name && vm.connectionName == connection);
+    if (!vm) {
         return (
             <>
-                {this.getInlineNotifications()}
-                {pathVms && <HostVmsList vms={vms}
-                    ui={ui}
-                    storagePools={storagePools}
-                    networks={networks}
-                    actions={vmActions} />
-                }
-                {path.length > 0 && path[0] == 'storages' && loggedUser &&
-                <StoragePoolList storagePools={storagePools}
-                    vms={vms}
-                    loggedUser={loggedUser}
-                    libvirtVersion={systemInfo.libvirtVersion} />
-                }
-                {path.length > 0 && path[0] == 'networks' &&
-                <NetworkList networks={networks} />
-                }
+                {getInlineNotifications()}
+                <EmptyStatePanel title={ cockpit.format(_("VM $0 does not exist on $1 connection"), name, connection) }
+                    action={_("Go to VMs list")}
+                    actionVariant="link"
+                    onAction={() => cockpit.location.go(["vms"])}
+                    icon={ExclamationCircleIcon} />
+            </>
+        );
+    } else if (vm.isUi && vm.createInProgress) {
+        return (
+            <>
+                {getInlineNotifications()}
+                <EmptyStatePanel title={cockpit.format(vm.downloadProgress ? _("Downloading image for VM $0") : _("Creating VM $0"), name)}
+                    action={_("Go to VMs list")}
+                    actionVariant="link"
+                    onAction={() => cockpit.location.go(["vms"])}
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2412-f8154e83-20251121-170542-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)